### PR TITLE
fix(closeout): harden worktree terminal completion

### PR DESCRIPTION
**Pull request by jamesryancooper**
Policy reference: `.octon/framework/execution-roles/practices/pull-request-standards.md`
Reminder: Run `@codex review`.

## What

Hardens `closeout-worktree` terminal completion validation so closed
candidates must cite completed, route-specific `closeout-change` receipts and
explicit residue routing classes.

Adds classifier routing counts, terminal-state negative controls, aligned
framework/Codex skill projection docs, and regenerated effective runtime,
capability, extension, and run-health evidence needed for hosted freshness
checks.

## Why

No-Issue: operator-requested implementation of closeout-worktree terminal
completion hardening.

The wrapper previously had remaining gaps where unsafe residue, missing
lifecycle proof, deferred cleanup, ordinary untracked residue, or recursive
closeout evidence loops could still cause premature stop or false terminal
claims.

## How

The wrapper validator now requires candidate-keyed `residue_routing_class`
values, verifies branch/direct route lifecycle facts from cited receipts, and
rejects `nonterminal` reports with no genuine unresolved condition.

Residue classification now keeps legacy counters while adding
publishable/private/foreign/unsafe/ambiguous routing counts. Tests cover valid
terminal paths and fail-closed cases for missing authorization, missing
exact-SHA checks, raw/private state, `.octon/engine/**`, deferred cleanup, and
recursive final evidence publication.

Generated effective-state artifacts were refreshed after the implementation so
the PR's hosted architecture, runtime, and mission checks evaluate current
publication receipts instead of stale receipt references.

## Profile Selection Receipt

- Semantic version source(s): workspace charter and constitutional charter
  declare `release_state: pre-1.0`.
- Release state (`pre-1.0` or `stable`): `pre-1.0`.
- `change_profile` (`atomic` or `transitional`): `atomic`.
- Hard-gate facts (downtime, coordination, migration/backfill, rollback, blast
  radius, compliance): no downtime, no migration/backfill, no dependency
  change, no release activation, no data movement, no compatibility break to
  existing low-level counters.
- Tie-break status: no profile tie-break required.
- `transitional_exception_note` (required when `pre-1.0` + `transitional`):
  n/a.

## Implementation Plan

- Workstreams and scope: harden wrapper validation, extend residue classifier,
  update wrapper docs/projections, update tests, refresh generated effective
  publication state.
- Public interfaces/types/contracts affected: closeout-worktree report
  semantics add required `residue_routing_class`; legacy classifier counts
  remain compatible.
- Test scenarios: valid multi-candidate terminal completion, valid retained
  residue, invalid missing no-PR authorization/check evidence, invalid false
  nonterminal, invalid unsafe/raw residue, invalid recursive closeout evidence
  loop, generated freshness validation.
- Assumptions/defaults: no proposal packet, no default-work-unit replacement,
  no direct wrapper Git mutation.

## Impact Map (code, tests, docs, contracts)

- Code: `.octon/framework/assurance/runtime/_ops/scripts/validate-closeout-worktree-wrapper.sh`,
  `.octon/framework/assurance/runtime/_ops/scripts/classify-change-closeout-residue.sh`.
- Tests: `.octon/framework/assurance/runtime/_ops/tests/test-closeout-worktree-wrapper.sh`.
- Docs/skills: framework and Codex closeout-worktree skill docs plus related
  host projections for closeout-change and repo-hygiene cleanup evidence
  boundaries.
- Generated/evidence: effective capability, extension, runtime route-bundle,
  support reconciliation, run-health read models, and referenced publication
  receipts.
- Contracts: no schema file changed; report semantics and validation docs
  updated.

## Compliance Receipt

- [x] Selected exactly one execution profile before planning/implementation.
- [x] Applied release-maturity gate from semantic version.
- [x] Enforced pre-1.0 default (`atomic`) unless hard gates required
  `transitional`.
- [x] Included `transitional_exception_note` when required.
- [x] Included tie-break escalation when applicable.
- [x] Updated impacted contracts/docs/tests.
- [x] Honored charter change-control constraint (no direct principles charter
  edits without override evidence).

## Exceptions/Escalations

none

## Minimality / Anti-Bloat Receipt

- Existing surfaces searched: closeout-worktree skill docs, closeout-change
  lifecycle contracts, default-work-unit policy, residue classifier, wrapper
  validator, repo-hygiene governance, evidence disclosure tiers, host
  projections, generated effective-state validators.
- Existing utilities reused: shell validators, current wrapper test harness,
  publication generators, run-health generator, and cleanup authorization
  helper flow; no new helper framework added.
- New files and abstractions: no new abstractions; new files are generated or
  receipt evidence from existing publication routes.
- Dependency changes: none.
- Cleanup pass result: cleanup dry-run reported `cleanup_candidates: 0` and
  `manual_review: 0`; retained untracked evidence was protected because it is
  referenced by tracked control, generated, governance, or evidence files.
- Behavior-preservation evidence: legacy classifier `classes:` counters remain
  in place; routing counts are additive.
- Generated/input/proposal/authority-boundary checks: raw `.octon/state/**`,
  `.octon/engine/**`, generated authority, and input surfaces fail closed
  unless candidate-specific routing proves otherwise.
- Remaining implementation-quality risk: hosted checks are still authoritative
  for branch-pr readiness and mergeability.

## Tradeoffs

The validator is intentionally stricter and may reject older wrapper reports
that omit routing classes. That is expected for new valid reports because
terminal completion now depends on explicit residue routing proof.

The generated freshness commit is large because current publication receipts
and run-health projections must be coherent for hosted architecture and
runtime checks. It is separated from the implementation commit.

## Testing

- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-closeout-worktree-wrapper.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/tests/test-closeout-worktree-wrapper.sh` passed, 49/49.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-change-closeout-lifecycle-alignment.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/tests/test-change-closeout-lifecycle-alignment.sh` passed, 54/54.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-default-work-unit-alignment.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-repo-hygiene-governance.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-evidence-disclosure-tiers.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-capability-publication-state.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-extension-publication-state.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/tests/test-mission-autonomy-scenarios.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-effective-state.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-effective-route-bundle.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-support-envelope-reconciliation.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-run-health-read-model.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-architecture-conformance.sh` passed.
- `bash .octon/framework/assurance/runtime/_ops/scripts/cleanup-local-run-artifacts.sh` dry-run reported `cleanup_candidates: 0`, `manual_review: 0`.
- `git diff --check` passed.
- `git diff --cached --check` passed before the generated freshness commit.

## Rollout

Merge as one atomic governance/runtime hardening change after PR checks pass.
No flag or migration required.

## Convivial Purpose Check

- [x] Feature expands genuine user capability (not synthetic engagement).
- [x] Attention and interruption behavior are justified and user-controllable.
- [x] No manipulative patterns or dark-pattern mechanics are introduced.
- [x] Data collection/extraction risk is minimal and explicitly justified.

## Risk Rubric

- Risk class: [ ] Trivial [ ] Low [ ] Medium [x] High
- Rollback plan: revert the PR squash commit or discard the branch before
  merge.
- Rollback handle: branch
  `chore/OCTON-000-closeout-worktree-terminal-hardening`, commits
  `bab4e777ec5fd2dd847eb339c8838864e57abfbc`,
  `f56d465d0b8e6ac1e3f373578992986037a8d9ed`, and
  `16f2fa8cd7b6ce58b49551a9fd32e7d5979b2bf9`, plus receipt refresh
  `dd161db85f3e390367b8a7a8471dd99e60d1424a`.
- Flags changed (name, owner, expiry, rollout): none.
- Autonomy eligibility: [ ] autonomy:auto-merge [x] autonomy:no-automerge

## Contracts and Threat Model

- OpenAPI/JSON-Schema changes: none.
- Threat model update/link: terminal-closeout false-positive and unsafe-residue
  risks addressed in validator tests.

## Observability and Performance

- Traces/logs/metrics for changed flows: validator output is the observable
  signal.
- Representative traces for high risk changes: local validation output listed
  under Testing.
- Performance or bundle impact: none.

## License and Provenance

- New dependencies and licenses: none.
- Generated code/templates provenance notes: host skill projections updated
  from framework skill content; generated effective-state artifacts and
  publication receipts were produced by existing repo-local generators.

## Checklist

- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [ ] Reviewer feedback addressed with fix, commit, push, reply; reviewer-owned threads confirmed by reviewer/maintainer or resolved under the documented solo-maintainer exception

## Screenshots / Notes

No visual changes. Latest hosted checks are pending after commit
`dd161db85f3e390367b8a7a8471dd99e60d1424a`; PR remains draft until required
checks and review gates are green.

### CODE REVIEW

File: .octon/framework/assurance/runtime/_ops/scripts/validate-closeout-worktree-wrapper.sh [hunk 1]
```diff
@@ -1024,13 +1318,35 @@ if selected_candidate_disposition in {"blocked", "escalated", "ambiguous"} or se
...
+    for candidate_id, state in final_states.items()
+    if state in {"blocked", "escalated", "deferred"}
+}
+for candidate in unresolved_candidates:
+    candidate_id = candidate.get("candidate_id")
+    if candidate.get("disposition") in deferred_dispositions | blocker_dispositions and is_nonempty_string(candidate_id):
+        genuinely_unresolved_candidate_ids.add(candidate_id)
+for candidate_id, items in blockers_by_candidate.items():
+    if items:
+        genuinely_unresolved_candidate_ids.add(candidate_id)
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/assurance/runtime/_ops/scripts/validate-closeout-worktree-wrapper.sh:1349]: Fixed in ccc823847: blockers whose candidate_id is not a declared candidate now fail validation and are not added to blockers_by_candidate, so stray blockers cannot satisfy nonterminal proof. Added test coverage in case_nonterminal_with_unknown_candidate_blocker_fails.
